package/ios/RNSkia-iOS/RNSkMetalCanvasProvider.h

```diff
@@ -30,14 +39,16 @@ class RNSkMetalCanvasProvider : public RNSkia::RNSkCanvasProvider {
 
   CALayer *getLayer();
 
+  static sk_sp<SkSurface> MakeOffscreenMetalSurface(int width, int height);
```

Comment:
I am not sure what the first comment means? Could you elaborate on the change you would like to be considered here?

SkiaMetalHelper doesn't feel sensible because our goal with SkiaOpenGLHelper is essentially an helper so we have use the OpenGL API with the same level of convenience that we already have using Metal.

Reply:
I don't think it looks correct that RNSkMetalCanvasProvider should have a static MakeOffscerenMetalSurface factory method so following the pattern from Android of having a separate factory class makes it a cleaner and easier to understand.

I agree that we don't necessarily need a helper here - just thought it could help hide some of the complexities of the Metal setup with all id<MtlXXX>, retain, release etc stuff that might be a bit hard to understand.

I think the main argument here is that we found a very clear and understandable way of structuring the classes on Android after spending a lot of time on the design - iOS would benefit from this as well - making it easier to maintain and develop in the future as well so we should take the time to align the two platforms.

Reply:
I see it yes.